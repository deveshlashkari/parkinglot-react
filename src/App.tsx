//@ts-nocheck
import React from 'react'
import './App.css';
import { BrowserRouter as Router } from "react-router-dom"
import AppRotue from './AppRoutes'

const App = () => {

    return (
      <Router>
       <AppRotue/>  
      </Router>
    );
}

export default App;
