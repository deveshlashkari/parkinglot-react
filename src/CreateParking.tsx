//@ts-nocheck
import react, { useState } from 'react'
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';

const CreateParking = ({ numOfParking, setNumOfParking, CreateParkingLots, error }: Props) => {

    return (
        <div style={{ textAlign: "center", display: "inline-grid" }}>
            <h2>Create Prackings</h2>
            <TextField id="outlined-basic"
                label="Number of parking lot"
                variant="outlined"
                type="number"
                value={numOfParking}
                onChange={(e) => {
                  setNumOfParking(e.target.value)
                }}
            />
            {error&&"Limit is exceed"}
            <Button
                data-test-id="btn-create"
                style={{background: !numOfParking? "#acbfe0": "cornflowerblue",  marginTop: 15, color:"black"}}     
                disabled={!numOfParking}
                onClick={ CreateParkingLots}>Submit
            </Button>
        </div>
    )
}

export default CreateParking