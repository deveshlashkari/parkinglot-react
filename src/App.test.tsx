import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer';
import CreateParking from './CreateParking'
import { mount } from 'enzyme';

it('test snap-shot ', () => {  
  const CreateParkingComponent = renderer.create(<CreateParking />).toJSON();
  expect(CreateParkingComponent).toMatchSnapshot();
});

it('create parking slots', () => {
  const component = renderer.create(
    <CreateParking>Create Prackings</CreateParking>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

test('render button', () => {
  const {getByText} = render(<CreateParking/>)

  const submit = getByText(/btn-create/i);
  expect(submit).toBeInTheDocument()
})


