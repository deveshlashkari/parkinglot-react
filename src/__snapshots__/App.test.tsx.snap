// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create parking slots 1`] = `
<div
  style={
    Object {
      "display": "inline-grid",
      "textAlign": "center",
    }
  }
>
  <h2>
    Create Prackings
  </h2>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="outlined-basic"
      id="outlined-basic-label"
    >
      Number of parking lot
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="outlined-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1ftyaf0"
        >
          <span>
            Number of parking lot
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-test-id="btn-create"
    disabled={true}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "background": "#acbfe0",
        "color": "black",
        "marginTop": 15,
      }
    }
    tabIndex={-1}
    type="button"
  >
    Submit
  </button>
</div>
`;

exports[`test snap-shot  1`] = `
<div
  style={
    Object {
      "display": "inline-grid",
      "textAlign": "center",
    }
  }
>
  <h2>
    Create Prackings
  </h2>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="outlined-basic"
      id="outlined-basic-label"
    >
      Number of parking lot
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="outlined-basic"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1ftyaf0"
        >
          <span>
            Number of parking lot
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    data-test-id="btn-create"
    disabled={true}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "background": "#acbfe0",
        "color": "black",
        "marginTop": 15,
      }
    }
    tabIndex={-1}
    type="button"
  >
    Submit
  </button>
</div>
`;
